### PR TITLE
tools: use os home dir to instead of the hardcoded PATH

### PR DESCRIPTION
**Pull request by JayT106**
porting PR #6498 to main

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: tools/tm-signer-harness/main.go [hunk 1]
```diff
@@ -59,6 +60,14 @@ Use "tm-signer-harness help <command>" for more information about that command.`
 		fmt.Println("")
 	}
 
+	hd, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Println("The UserHomeDir is not defined, setting the default TM Home PATH to \"~/.tendermint\"")
+		defaultTMHome = "~/.tendermint"
+	} else {
+		defaultTMHome = fmt.Sprintf("%s/.tendermint", hd)
```
Comment (cmwaters) [inline tools/tm-signer-harness/main.go:68]: nit: Use `filepath.Join(hd, ".tendermint")`

**Review by cmwaters — APPROVED** [review]

File: tools/tm-signer-harness/main.go [hunk 2]
```diff
@@ -59,6 +60,14 @@ Use "tm-signer-harness help <command>" for more information about that command.`
 		fmt.Println("")
 	}
 
+	hd, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Println("The UserHomeDir is not defined, setting the default TM Home PATH to \"~/.tendermint\"")
+		defaultTMHome = "~/.tendermint"
```
Comment (williambanfield) [inline tools/tm-signer-harness/main.go:66]: It looks like this is just expanded with the `user.Current()` call here https://github.com/tendermint/tendermint/blob/21a3bbda3fd8d272cc7fee888b4339a669cbcaad/tools/tm-signer-harness/main.go#L110
https://github.com/tendermint/tendermint/blob/21a3bbda3fd8d272cc7fee888b4339a669cbcaad/tools/tm-signer-harness/internal/utils.go#L13

Not sure I see what benefit this brings?

**Comment by thanethomson** [comment]
@JayT106 do you still use the signer harness?

**Comment by JayT106** [comment]
> @JayT106 do you still use the signer harness?

No. If the tendermint doesn't support `signer harness` anymore, we can close this PR

File: tools/tm-signer-harness/main.go (hunk 2 above)
Comment (JayT106) [reply tools/tm-signer-harness/main.go]: didn't notice there is a util function for it. just updated.

File: CHANGELOG_PENDING.md [hunk 3]
```diff
@@ -17,6 +17,9 @@
 - Data Storage
   - [state] \#6541 Move pruneBlocks from consensus/state to state/execution. (@JayT106)
   
+- Tooling
+  - [tools] \#6498 Set OS home dir to instead of the hardcoded PATH. (@JayT106)
```
Comment (thanethomson) [inline CHANGELOG_PENDING.md:21]:
```suggestion
  - [tools/tm-signer-harness] \#6498 Set OS home dir to instead of the hardcoded PATH. (@JayT106)
```

**Review by thanethomson — APPROVED** [review]
Let's keep this for now.

**Pull request merged**